### PR TITLE
Update Enrollment failed message with status code

### PR DESCRIPTION
**Pull request by ahsan-ul-haq**
@rlucioni currently I just added status code with the error message. Is there any other information you think we should add.  

cc: @schenedx 

ECOM-4763

### CODE REVIEW

**Comment by rlucioni** [comment]
@ahsan-ul-haq please include a ticket number (i.e., ECOM-4763) in your commit message.

**Comment by ahsan-ul-haq** [comment]
@rlucioni please review

File: ecommerce/extensions/fulfillment/modules.py [hunk 1]
```diff
@@ -231,8 +231,8 @@ def fulfill_product(self, order, lines):
                         reason = '(No detail provided.)'
 
                     logger.error(
-                        "Unable to fulfill line [%d] of order [%s] due to a server-side error: %s", line.id,
-                        order.number, reason
+                        "Fulfilling line [%d] of order [%s] failed with status code %d due to a server-side error: %s",
```
Comment (rlucioni) [inline ecommerce/extensions/fulfillment/modules.py]: The error message shouldn't assume a server-side error (the error could be on the client, a 400). Please change this to: `Fulfillment of line [%d] on order [%s] failed with status code [%d]: %s`

**Review by rlucioni — CHANGES_REQUESTED** [review]
@ahsan-ul-haq please include a JIRA ticket number in your _Git commit_ message. This is not the same as the first comment on a GitHub PR.

**Comment by ahsan-ul-haq** [comment]
@rlucioni made the changes suggested

**Review by schenedx — APPROVED** [review]
👍

**Review by rlucioni — APPROVED** [review]
Approved once you squash and update your commit message.

**Pull request merged**